eng/packaging.targets

```diff
@@ -288,4 +288,19 @@
     <Error Condition="'$(ServicingVersion)' == '0'" Text="ServicingVersion is set to 0 and it should be an increment of the patch version from the last released package." />
   </Target>
 
+  <ItemDefinitionGroup>
+    <TargetPathWithTargetPlatformMoniker>
+      <GeneratePackageOnBuild>$(GeneratePackageOnBuild)</GeneratePackageOnBuild>
+    </TargetPathWithTargetPlatformMoniker>
+  </ItemDefinitionGroup>
+
+  <Target Name="ValidateBuiltPackageInServicing" 
+          Condition="'$(PreReleaseVersionLabel)' == 'servicing' and '$(IsPackable)' == 'true'"
+          AfterTargets="ResolveReferences">
+
+      <Error Condition="'$(GeneratePackageOnBuild)' != 'true' and 
+                        '%(ReferencePath.ReferenceSourceTarget)' == 'ProjectReference' and 
+                        '%(ReferencePath.GeneratePackageOnBuild)' == 'true'"
+             Text="This project did not set GeneratePackageOnBuild, but dependencies '%(ReferencePath.OriginalProjectReferenceItemSpec)' did.  Please ship this project by setting GeneratePackageOnBuild to true." />
```

Comment:
We could even do something similar to this for telling people to set `GeneratePackageOnBuild` in the first place.  If we could see the files they touched in their PR, intersect those with the input to the compiler, and if any were found error we'd have PR validation for the case where folks forgot to set `GeneratePackageOnBuild`.

Reply:
If we come to place where we are able to detect that, (and thinking a bit outside of the box here) would it be possible to get rid of GeneratePackageOnBuild all together and calculate it on the fly? This would reduce mistakes on servicing PRs of only editing the code but forgetting to set the property, and I'm thinking we would have some sort of checked in file that gets modified by the build (similar to what we used to have with packageIndex.json) where we would have the build add entries there of packages that have been edited for this servicing release, and then branding PRs would only go and empty that file to go back to not generate any package.